### PR TITLE
Add tests for all derived instances

### LINKED ISSUE #425: [Scala 3] - Wildcard derivation import not working
Super simple reproduction:

```scala
import cats.Show
// import cats.derived.semiauto.{ derived, product } // this works
// import cats.derived.semiauto.{ given, * } // this works (as per the 3rd comment)
import cats.derived.semiauto.* // does not work

enum TradeAction derives Show:
  case Ask, Bid
```

It fails with the following error:

```
no implicit argument of type cats.Show[(Playground.TradeAction.Ask : Playground.TradeAction)] was found

The following import might fix the problem:

  import cats.derived.semiauto.product
```

https://scastie.scala-lang.org/0q4XaUbJTqibfe78MuyzAA

---

Not sure whether this has already been discussed or not but I figured it might be better to keep track of it right here.

### PR DESCRIPTION
**Pull request by joroKr21**
Unfortunately there is some code duplication because I didn't find a way to make this generic.
But the boilerplate is minimal because we just wrap existing ADTs into a sort of newtype case class.

Closes #425 by making sure it works for all type classes

### CODE REVIEW

File: core/src/test/scala-3/cats/derived/ADTs.scala [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+        Arbitrary.arbitrary[Int].map(LeafI.apply)
+      )
+    )
+
+    given Cogen[EnumK0] = Cogen[Either[String, Int]].contramap {
+      case LeafS(s) => Left(s)
+      case LeafI(i) => Right(i)
+    }
+
+  enum EnumK1[A]:
```
Comment (TimWSpence) [inline core/src/test/scala-3/cats/derived/ADTs.scala:43]: Was this meant to be Covariant?

File: core/src/test/scala-3/cats/derived/KittensSuite.scala [hunk 2]
```diff
@@ -16,18 +16,23 @@
...
+import org.scalacheck.{Arbitrary, Cogen}
 import org.scalacheck.Test.Parameters
 
+import scala.deriving.Mirror
 import scala.quoted.*
 
 /** An opinionated stack of traits to improve consistency and reduce boilerplate in Kittens tests. Note that unlike the
   * corresponding CatsSuite in the Cat project, this trait does not mix in any instances.
   */
-abstract class KittensSuite extends KittensSuite.WithoutEq, TestEqInstances
```
Comment (TimWSpence) [inline core/src/test/scala-3/cats/derived/KittensSuite.scala:30]: What are these instances used for? I thought we were defining explicit instances for everything in `ADTS`?

File: core/src/test/scala-3/cats/derived/ADTs.scala [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+
+package cats.derived
+
+import cats.{Eq, Eval}
+import cats.syntax.all.given
+import org.scalacheck.rng.Seed
+import org.scalacheck.{Arbitrary, Cogen, Gen}
+import scala.annotation.tailrec
+
+object ADTs:
```
Comment (TimWSpence) [inline core/src/test/scala-3/cats/derived/ADTs.scala:25]: So are we dropping testing for sealed traits in favour of enums? I had assumed that we would test both (although that sucks 😂 )

**Review by TimWSpence — APPROVED** [review]
This is awesome! Thanks so much!!

File: core/src/test/scala-3/cats/derived/KittensSuite.scala (hunk 2 above)
Comment (joroKr21) [reply core/src/test/scala-3/cats/derived/KittensSuite.scala:30]: We have the `EqInstances` separately to be able to test `Eq` without interference 🐔 🥚

File: core/src/test/scala-3/cats/derived/ADTs.scala (hunk 3 above)
Comment (joroKr21) [reply core/src/test/scala-3/cats/derived/ADTs.scala:25]: We're not dropping anything 🤔 - we just add more tests

File: core/src/test/scala-3/cats/derived/ADTs.scala (hunk 1 above)
Comment (joroKr21) [reply core/src/test/scala-3/cats/derived/ADTs.scala:43]: I think it doesn't matter

**Comment by gvolpe** [comment]
Thank you for all this work 😍 ! New milestone coming out soon?

File: core/src/test/scala-3/cats/derived/ADTs.scala (hunk 3 above)
Comment (TimWSpence) [reply core/src/test/scala-3/cats/derived/ADTs.scala:25]: Oh so we are! 🤦

**Comment by TimWSpence** [comment]
> Thank you for all this work 😍 ! New milestone coming out soon?

We're aiming for at least an RC if not full release ASAP :)

**Pull request merged**